src/custom-blocks/featured-item/edit.js

```diff
@@ -0,0 +1,380 @@
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+const { Fragment } = wp.element;
+const d = new Date();
+const allowedMediaTypes = ['image'];
+const templatefeaturedItemBlock = [
+	[ 'core/heading', { placeholder: 'Add featured item section title' } ]
+];
+const noItemSelectedText = "No item selected"; // Text in editor to shew that no item will be displayed
+
+export default function featuredItemEdit({ attributes, setAttributes} ) {
+
+	const {
+		featuredImage,
+		featuredItemType,
+		featuredItemID,
+		featuredItemHasDate,
+		featuredItemHasBar,
+		featuredLinkText,
+		featuredCustomImage,
+		className,
+	} = attributes;
+
+	const {
+		allPostTypes,
+	} = useSelect(
+		( select ) => {
+
+			const { getPostTypes } = select(
+				coreStore
+			);
+			const allPostTypeThingies = getPostTypes(
+				{ per_page: -1 }
+			);
+			return {
+				allPostTypes: allPostTypeThingies
+			};
+		}
+	);
+
+	const {
+		allDocuments,
+	} = useSelect(
+		( select ) => {
+			if (featuredItemType != "" && featuredItemID.length > 0) {
+
+				const { getEntityRecords } = select(
+					coreStore
+				);
+				const { getSettings } = select( blockEditorStore );
+				const { imageSizes, imageDimensions } = getSettings();
+
+				const posts = getEntityRecords(
+					'postType',
+					featuredItemType,
+					{ per_page: 24 } // restricts to the 25 options (including none)
+				);
+
+				return {
+					allDocuments: posts
+				};
+			} else {
+				return {
+					allDocuments: false
+				};
+			}
+		}
+	);
+
+	let itemTypes = [{
+		label: "-",
+		value: ""
+	}]
+
+	if (allPostTypes) {
+		allPostTypes.forEach(thisPostType => {
+			if (thisPostType.name != "Posts" && thisPostType.name != "Pages" && thisPostType.name != "Media" && thisPostType.viewable) {
+				itemTypes.push({
+					label: thisPostType.name,
+					value: thisPostType.slug
+				});
+			}
+		});
+	}
+
+	let docOptions = [
+		{ label: "None", value: '0' },
+	]
+
+	var docList = [
+		{ title: noItemSelectedText, summary: "", date: "date", image: "", imageID: "0"}
+	];
+
+	if (Array.isArray( allDocuments )) {
+		for (let i=0;i<allDocuments.length;i++) {
+			docList[allDocuments[i].id] = {
+				title: allDocuments[i].title.rendered,
+				summary: allDocuments[i].post_meta.summary,
+				date: allDocuments[i].date,
+				imageID: allDocuments[i].featured_media,
+				image: allDocuments[i].featured_image_url,
+			}
+			docOptions.push({label: allDocuments[i].title.rendered, value: allDocuments[i].id});
+
+		}
+	}
+
+	/* Not used yet
+	const changeImage = imageObject => {
+		var imageSizes = imageObject.sizes;
+		var image = imageSizes.large.url;
+		setAttributes({ featuredCustomImage: image })
+	}
+	*/
+
+	const removeImage = () => {
+		setAttributes({
+			featuredCustomImage: null,
+		});
+	};
+
+	const setItemType = newItemType => {
+		setAttributes({ featuredItemType: newItemType });
+	};
+
+	const setDocument = newDocumentID => {
```

Comment:
setFeaturedItem

Reply:
Changed.  Also changed other references to "document" such as allDocuments